puca_dsp-esp-idf/esp-idf_puca_dsp_setup/main/Audio.cpp

```diff
@@ -25,8 +25,11 @@ void Audio::I2S_init(void)
 
     i2s_driver_install(I2S_NUM_0, &i2s_config, 0, NULL);                            // install the driver on Core 0
     i2s_set_pin(I2S_NUM_0, &pin_config);                                            // port & pin config 
-    PIN_FUNC_SELECT(PERIPHS_IO_MUX_GPIO0_U, FUNC_GPIO0_CLK_OUT1);                   // MCLK on GPIO0
-    REG_WRITE(PIN_CTRL, 0xFFFFFFF0);                                                
+    // Deprecated pin setting for clk setup:
+    // PIN_FUNC_SELECT(PERIPHS_IO_MUX_GPIO0_U, FUNC_GPIO0_CLK_OUT1);                   // MCLK on GPIO0
+    // REG_WRITE(PIN_CTRL, 0xFFFFFFF0);                                                
+    uint32_t bits_cfg = (I2S_BITS_PER_CHAN_32BIT << 16) | I2S_BITS_PER_SAMPLE_16BIT;
+    i2s_set_clk(I2S_NUM_0, 22050, bits_cfg, I2S_CHANNEL_STEREO);
```

Comment:
In `i2s_set_clk`, `22050` should be `32000` (the sample rate in this example), or maybe better `I2S_SAMPLE_RATE` ?

Reply:
Yes good point! Have implemented this change.